tests/test_proxy_digest_auth.py

```diff
@@ -15,26 +16,88 @@ def setUp(self):
         self.auth = http_proxy_digest.HTTPProxyDigestAuth(
             self.username, self.password
         )
-        self.auth.last_nonce = "bH3FVAAAAAAg74rL3X8AAI3CyBAAAAAA"
-        self.auth.chal = {
-            'nonce': self.auth.last_nonce,
-            'realm': 'testrealm@host.org',
-            'qop': 'auth'
-        }
         self.prepared_request = requests.Request(
             'GET',
             'http://host.org/index.html'
         ).prepare()
 
-    def test_proxy_digest(self):
+    def test_with_existing_nonce(self):
         """Test if it will generate Proxy-Authorization header
-        when nonce presents.
+        when nonce present.
         Digest authentication's correctness will not be tested here.
         """
+        self.auth.last_nonce = "bH3FVAAAAAAg74rL3X8AAI3CyBAAAAAA"
+        self.auth.chal = {
+            'nonce': self.auth.last_nonce,
+            'realm': 'testrealm@host.org',
+            'qop': 'auth'
+        }
+
         # prepared_request headers should be clear before calling auth
         assert not self.prepared_request.headers.get('Proxy-Authorization')
         self.auth(self.prepared_request)
         assert self.prepared_request.headers.get('Proxy-Authorization')
 
+    def test_no_challenge(self):
+        """Tests that a response containing no authentication challenge is
+        left alone.
+        """
+        connection = MockConnection()
+        first_response = connection.make_response(self.prepared_request)
+        first_response.status_code = 404
+
+        assert not self.auth.last_nonce
```

Comment:
`assert not` is a bit ... vague. Could this be a bit more specific, e.g.,

```py
assert self.auth.last_nonce is None
```

That's much more explicit.